src/main/resources/com/google/api/codegen/csharp/gapic_smoketest.snip

```diff
@@ -0,0 +1,70 @@
+@extends "csharp/common.snip"
+@extends "csharp/common_initcode.snip"
+
+
+# smoke: SmokeTestClassView
+# smoke.method: TestCaseView
+@snippet generate(smoke)
+    {@renderFileHeader(smoke.fileHeader)}
+
+    namespace {@smoke.fileHeader.packageName}.SmokeTests
+    {
+        public class {@smoke.name}
+        {
+            public static int Main(string[] args)
+            {
+                @if smoke.requireProjectId
+                    // Read projectId from args
+                    if (args.Length != 1)
+                    {
+                        Console.WriteLine("Usage: Project ID must be passed as first argument.");
+                        Console.WriteLine();
+                        return 1;
+                    }
+                    string projectId = args[0];
+
+                @end
+                // Create client
+                {@smoke.apiClassName} client = {@smoke.apiClassName}.Create();
+
+                @if smoke.hasMethod
```

Comment:
Do we only support a single test method per service?

Reply:
Apparently yes, only a single test per service.